### DISCUSSION TITLE
How to treat time for an abundance index

### DISCUSSION THREAD

**Discussion by Joseph-Barss** [sdmTMB/sdmTMB#359] [Q&A] [answered]
Hello,

I have a question about the best way to make an abundance index using sdmTMB.

My research involves fitting a species distribution model to a marine species, and I am using sdmTMB to do this. After fitting my candidate models, I use the get_index() function to look at the estimated abundance in the region. It seems that it is common to treat the year as a categorical factor in the model when making an index, but I just wanted to ask if this is done for theoretical or computational reasons. Is it possible/a good idea to model the effect of year as a random walk?

In addition, I am curious to know whether the correlation structure of the spatiotemporal random fields should match that of the year term. For example, if I do treat year as a factor, should I only use iid spatiotemporal random fields?

Thank you in advance for your help!

Joseph

**Comment by seananderson** [comment, marked as answer]
Hi Joseph,

For use in stock assessment models, the standard has typically been to use categorical year effects and IID spatiotemporal fields. This (1) is the most assumption-free model that avoids constraining/smoothing the index before the assessment model tries to reconcile plausible dynamics and (2) reduces correlation between the annual estimates given this correlation is ignored in most assessment models. Also, the AR(1) process is mean-reverting, so your index could start reverting to the mean if data are sparser or non-existent in some years. This wouldn't be the case for the random walk.

There are cases where you may want to deviate from that standard.

1. You may not be preparing an index for an assessment and instead want to produce a more biologically realistic somewhat smoothed index. Or perhaps your intent for fitting the model is not the index, but inference on some quantity and so you just want to allow the mean to vary through time. Or perhaps you want to interpolate across missing years or project into the future. In these cases, you might consider a random walk or AR(1) process on the year intercepts. For interpolation or projection, you'd need an autoregressive process on the mean or the fields.

2.  I have seen situations when some fairly extreme deviations from a class sampling design (e.g., surveying half the survey domain in even years and the other half in odd years or dramatically expanding the survey domain or adding in an additional survey with a different spatial domain in some years) combined with a gradient in population density where some degree of penalizing time variability is necessary to avoid the model erroneously placing spatial variability into time. E.g., the model might think all even years are high and all odd years are low. This is a problem I'm actively working on.

> In addition, I am curious to know whether the correlation structure of the spatiotemporal random fields should match that of the year term. For example, if I do treat year as a factor, should I only use iid spatiotemporal random fields?

That's up to you and depends what you want to happen with the mean vs. the spatiotemporal latent effects. I could see reasons for various combinations. I've certainly seen independent years with AR1 fields to allow the mean to vary unconstrained but allow for persistent hotspots in space (usually in ecological studies vs. index standardization for assessment). An AR(1) can, of course, end up estimated with a correlation close to zero (same as IID) or close to 1 (approaching a random walk).

**Reply by Joseph-Barss** [reply]
Thank you very much for your reply! I do have a few years that sample within a smaller subset of the area, but I have not noticed the issues mentioned in case 2 yet. I will keep an eye out for them, though. I don't believe I will investigate predictions for future years, so I will stick with categorical year effects. As for the random fields, I tested AR(1) and got a correlation of 0.98, which led me to try a random walk as well. Because both of these options lead to longer fitting times than IID fields, I have been reluctant to use them when testing other model components. Based on your reply, I think I will use a categorical year effect and IID random fields for most of my analyses, and then test the different types of spatiotemporal fields on the final model.

**Reply by kellijohnson-NOAA** [reply]
I am not sure what kind of run times you are talking about but I would advocate for being patient and running the models that you think you should rather than investigating with the IID spatial field and doing a sensitivity to the random walk for the spatial field in the end. Computing power is pretty cheap so is running models overnight. 
Disclaimer -- I am not a developer of this package just a user.

**Reply by Joseph-Barss** [reply]
Thank you for your reply! Based on your suggestion, I now plan to use a random walk for the fields in the final versions of my analyses, because the evidence seems to point towards that being the best option. I am currently still exploring different model options and trying to set up a simulation study, however, so I am fitting a lot of models fairly informally. I found that using the sdmTMB function to fit a single model with IID fields takes only a few minutes (on my computer), but with random walk or AR(1) fields it can take over 2 hours. I also figured out yesterday how to use the optimized BLAS library mentioned in the package README, which I hope will improve matters.

**Comment by sdowd20** [comment]
@seananderson, I have a question about your response to this question as it relates to my study. 

I am working with humpback whale density data in the Northeast US from 2003-2018. I am interested in building spatiotemporal models to determine what 1) environmental factors drive humpback whale presence/density and 2) see if humpback whale density has increased or decreased in this area over the years. 

Based on your response to this question, I have built a spatiotemporal model with IID fields and year as a fixed effect with a delta_gamma() distribution. I used cross-validation (cv_spatial with block size of 2*the spatial autocorrelation distance) to find the best formula describing humpback whale density, testing share_range= TRUE and share_range= FALSE. I am finding that my models run fine and pass sanity checks only when year as a factor is not included in the model formula. When it is, there are numerous sanity checks that fail. I am wondering how I can understand if humpback whale density has increased over the years if I am not able to include year as a fixed effect? For spatial models only, including year as a factor works fine.
